### PR TITLE
Add custom CA support for S3 backend

### PR DESCRIPTION
**Pull request by moelsayed**
https://github.com/rancher/rancher/issues/19222

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -857,3 +883,30 @@ func readZipFile(zf *zip.File) ([]byte, error) {
 	defer f.Close()
 	return ioutil.ReadAll(f)
 }
+
+func readS3EndpointCA(endpointCA string) ([]byte, error) {
```
Comment (superseb) [inline main.go]: My suggestion would be to make this more explicit:

- Check if its a file and log
- Check if its base64 encoded and log

I would also add a check for a valid certificate before continuing, possibly outside this function

**Comment by alena1108** [comment]
Would this be compatible with the older version of rke? As we don't branch rke-tools, ideally all versions should be compatible across master and release/v0.1

File: main.go [hunk 2]
```diff
@@ -857,3 +884,48 @@ func readZipFile(zf *zip.File) ([]byte, error) {
...
+	if err == nil {
+		log.Debug("reading s3-endpoint-ca as a base64 string")
+	} else {
+		ca, err = ioutil.ReadFile(endpointCA)
+		log.Debugf("reading s3-endpoint-ca from [%v]", endpointCA)
+	}
+	return ca, err
+}
+
+func isValidCertificate(c []byte) bool {
```
Comment (superseb) [inline main.go]: Can we have this function return the actual error from x509?

**Comment by superseb** [comment]
Added one comment, similar to the one in RKE, else LGTM.

**Pull request merged**